srml/balances/src/mock.rs

```diff
@@ -106,18 +145,19 @@ impl ExtBuilder {
 		self
 	}
 	pub fn build(self) -> runtime_io::TestExternalities<Blake2Hasher> {
+		EXISTENTIAL_DEPOSIT.with(|v| *v.borrow_mut() = self.existential_deposit);
```

Comment:
I see the interest of having this configured here, and I think it might be fine, but we have to keep in mind that this is a bit wrong as someone could write a legit test which is now failing:
```rust
let a = ExtBuilder:....build();
let b = ExtBuilder:...build();
with_externalities(a, ...));
with_externalities(b, ...));
```
maybe we could set those associated const using another mock function like `fn set_associated_const(....)` so this shows that this is permanent and not contained into externalities.